### ISSUE TITLE
please update branch oldTrunk

### ISSUE THREAD

**Issue by jonassmedegaard** [OpenSmalltalk/opensmalltalk-vm#604]
The oldTrunk branch was last updated in 2016 and misses some 100 commits.

Please update to be in sync with http://squeakvm.org/svn/squeak/trunk/

**Comment by pbella** [comment]
tl;dr version:  please don't look at this repo for updates to the squeak-vm Debian package.

These are two independently maintained repos.  The squeakvm.org repo your link points to is for the legacy VM (i.e. what currently exists in the squeak-vm Debian package and will remain so going forward - this is the one you want) and this repo is for the newer, high performance VM (i.e. what I am working on packaging up for the opensmalltalk-vm Debian package).  While they are conceptually similar VMs, they are very different implementations, separate codebases with different maintainers.  That's why we want to create the opensmalltalk-vm Debian package as the repos/VMs will not be merging but both are relevant to Smalltalk users as they run different Smalltalk image files. (i.e. the users data)

I wouldn't blame you for being a bit confused by this as even Smalltalkers get confused by it from time to time!

**Comment by jonassmedegaard** [comment]
I am aware that this git is mainly about developing other flavors of Smalltalk than Squeak.

That said, it does seem like this project *secondarily* serves as a mirror of Squeak: [this](http://squeakvm.org/svn/squeak/trunk/?p=3745) is bit-for-bit identical to [this](https://github.com/OpenSmalltalk/opensmalltalk-vm/tree/oldTrunk).

I am the official maintainer of [squeak-vm in Debian](https://tracker.debian.org/pkg/squeak-vm) and want to modernize that package, but want to do so in several steps. It is painful for me to interact with the Subversion VCS espeially for other-than-HEAD checkouts, which is the reason for raising this issue:
Please update the `oldTrunk` branch of this repo (even if that branch has no real use for your main development work) - it is a mirror helpful for my work on debian packaging of Squeak.

**Comment by dtlewis290** [comment]
This is not a mirror of the http://squeakvm.org/svn/squeak/trunk repository.

A number of recent commits have been made to incorporate the Debian patches, and you will find these updates in the upstream repository. Where possible I included the Debian bug references and credits to the original patch authors in the commit notices:
------------------------------------------------------------------------
r3810 | lewis | 2021-09-04 14:54:34 -0400 (Sat, 04 Sep 2021) | 12 lines

Add a cmake/configure option to specify name of the installed squeak script:

  --scriptname=<script> install squeak run script as <prefix>/bin/<script>

For example, if --scriptname=squeakvm is specified, then the run script
will be installed as /usr/local/bin/squeakvm rather than /usr/local/bin/squeak.
A corresponding update will be provided for the opensmalltalk-vm build
configuration in order to provide a mechanism for avoiding name collisions
when multiple types of VM are installed in the standard Unix directory structure.
Does not affect naming of squeak.sh, which may be subject to similar name
space conflicts.

------------------------------------------------------------------------
r3806 | lewis | 2021-07-08 18:46:59 -0400 (Thu, 08 Jul 2021) | 7 lines

Debian distro support.
Add --link-shared-lib option to the platforms/unix/configure script. If
specified, use dynamic linking to platform -ljpeg and -lpcre, otherwise
use default behavior of linking to the old but stable code in platforms
directory. Debian distro maintainers can now specify --link-shared-lib
to enforce dynamic linking to the system libraries.

------------------------------------------------------------------------
r3805 | lewis | 2021-07-08 18:42:07 -0400 (Thu, 08 Jul 2021) | 12 lines

Debian patch to use system shared libraries rather than older but stable
copies in the platforms directory. Applied without modification, the next
update will restore the ability to static link against local source.

Description: Link against system shared PCRE and JPEG libraries
Author: Neil Williams <codehelp@debian.org>
Author: Julian Taylor <jtaylor.debian@googlemail.com>
Origin: other, https://bugs.debian.org/634240#33
Bug-Debian: http://bugs.debian.org/634240
Bug-Debian: http://bugs.debian.org/710367
Last-Update: 2018-09-22

------------------------------------------------------------------------
r3803 | lewis | 2021-07-04 18:07:48 -0400 (Sun, 04 Jul 2021) | 7 lines

Harvest a patch from the Debian squeak-vm distribution
Description: Add sa_restorer on HPPA
Author: John David Anglin <dave.anglin@bell.net>
Origin: other, https://bugs.debian.org/528774#12
Bug-Debian: https://bugs.debian.org/528774
Last-Update: 2018-09-26

------------------------------------------------------------------------
r3802 | lewis | 2021-07-04 17:58:14 -0400 (Sun, 04 Jul 2021) | 7 lines

Harvest a patch from the Debian squeak-vm distribution
Description: Add missing includes
Author: Petter Reinholdtsen <pere@hungry.com>
Origin: other, https://bugs.debian.org/567366#10
Bug-Debian: https://bugs.debian.org/567366
Last-Update: 2018-09-22


Fix missing KeyRelease events when multiple keys are depressed.
Reference Mantis 0007597 http://bugs.squeak.org/view.php?id=7597.
Rather than keep a single lastKey to remember the last previously
pressed key value, maintain an array size 256 of last key pressed values
indexed by X11 KeyCode. Works for any number of simultaneous keys.

**Comment by dtlewis290** [comment]
My apologies for the bold font above, it is accidental and I did not intend to shout :-)

**Comment by pbella** [comment]
After discussing with Dave, we're thinking the easiest solution all around would be to generate a tarball for squeakvm that we could host on the main squeak.org site.  We think pruning the contents down to ./platforms/Cross, ./platforms/unix, and ./src would give you what you need (i.e. eliminating the non-Linux and svn portions of the the tree)  Would this be acceptable to you?

**Comment by jonassmedegaard** [comment]
Hi David - thanks for your work on squeak-vm.

(hint about that "loud" styling: posts done directly (not via email) can be updated later -> click on the ellipsis)

I appreciate your incorporating patches from the Debian packaging, and apologize for not having passed them upstream myself as is arguably my duty..

It is generally helpful that you as upstream release known stable code as a tarball - that eases work for redistributors like Debian.

This issue is not, however, about easing access to _latest_ code for the squeak-vm project.  Instead, it is about easing access to intermediary snapshots _other_ than latest code.

Therefore this issue is orthogonal to availability of tarball for squeak-vm (unless you mean to issue multiple tarballs for various points of development).

**Comment by pbella** [comment]
If I'm understanding Dave's idea, we're talking about creating a script that would generate the tarball which would exist as part of the repo.  So it should be possible from that point forward (i.e. once the script exists) to (re)generate a tarball at any given commit as needed.  I think it would be possible retain the old tarballs (with versioned filenames) for at least some period of time (i.e. several years) as this isn't something that needs to happen for each commit so it should be a manageable number of archives.

Realistically, I think we'd need to do it now (and possibly a handful of more times as needed to resolve any issues) to get the updated package into Debian testing, again to get squeak-vm and opensmalltalk-vm playing nicely together[1] and then not likely again until we need to prepare for the next Debian stable (again, with potentially a few iterations to resolve any issues discovered during packaging)  Being largely in maintenance mode, squeakvm is a relatively slow moving code base so aside from the *rare* issue, I don't think it's necessary to try to keep the Debian package up to the minute.

Does this make sense?

[1] Just to keep things simple, it probably makes sense to defer that discussion until we get the squeak-vm package current again.  Also, it helps to keep the moving pieces to a minimum since I'm not yet to a 'final draft' version of the initial opensmalltalk-vm package.

**Comment by jonassmedegaard** [comment]
That sounds good and helpful to me. Thanks for clarifying.

**Comment by pbella** [comment]
Jonas:

I'm posting here since I didn't see an appropriate bug to comment on in the Debian bug tracker.  Dave has worked with the Squeak web admins and gotten the tarball published here: http://files.squeak.org/debian/tarball/Squeak-4.19.9-3814-src.tar.gz

How would be the most appropriate way to communicate updates like this in the future?

Thanks,
Phil
